numpy/lib/tests/test_format.py

```diff
@@ -451,6 +462,10 @@ def test_roundtrip_truncated():
         if arr.dtype != object:
             assert_raises(ValueError, roundtrip_truncated, arr)
 
+def test_file_truncated():
+    for arr in basic_arrays:
+        if arr.dtype != object:
+            assert_raises(ValueError, file_truncated, arr)
```

Comment:
I realize this file uses `assert_raises`, but let's prefer `with pytest.raises(ValueError, match=...):`.

Also, please make sure to only include the actual reading inside the with statement.  (with `match=` its pretty safe, but this way a `ValueError` elsewhere might mean the test does nothing easily): the `try/except` should be as specific as is easy.